apps/prairielearn/src/pages/publicAssessments/publicAssessments.ts

```diff
@@ -0,0 +1,45 @@
+import * as express from 'express';
+import asyncHandler from 'express-async-handler';
+
+import * as error from '@prairielearn/error';
+
+import { selectAssessments, selectCourseInstanceIsPublic } from '../../models/course-instance.js';
+import { selectCourseInstanceById } from '../../models/course-instances.js';
+import { selectCourseById } from '../../models/course.js';
+
+import { AssessmentRowSchema, PublicAssessments } from './publicAssessments.html.js';
+
+const router = express.Router({ mergeParams: true });
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    const course_instance_id = req.params.course_instance_id;
+    res.locals.course_instance = await selectCourseInstanceById(course_instance_id);
+    res.locals.course = await selectCourseById(res.locals.course_instance.course_id);
+    res.locals.urlPrefix = '/pl';
+
+    const isPublic = await selectCourseInstanceIsPublic(course_instance_id);
+    if (!isPublic) {
+      throw new error.HttpStatusError(404, 'Course instance not public.');
+    }
```

Comment:
Other pages like `publicQuestions.ts` have a check for the `question-sharing` feature flag. Do you think we should do the same here?

Reply:
oh yes, we should do that. 

Come to think of it, I think we have a bigger problem--I don't think that we have feature flag guards around any of the sharing properties in the JSON. We missed putting this in when we did the UI -> JSON migration. Now, if someone without sharing turned on did mark something as shared, no one would have been able to import it, because they are not able to choose a sharing name. But theoretically I think they could have done it.

Reply:
addressed for this PR in f48d4887d8